sonar-plugin-api/src/main/java/org/sonar/api/design/Dependency.java

```diff
@@ -23,9 +23,12 @@
 import org.apache.commons.lang.builder.HashCodeBuilder;
 import org.apache.commons.lang.builder.ToStringBuilder;
 import org.sonar.api.resources.Resource;
-import org.sonar.graph.Edge;
 
-public class Dependency implements Edge<Resource> {
+/**
+ * @deprecated since 5.2 No more design features
+ */
+@Deprecated
+public class Dependency {
```

Comment:
:new: ![INFO](https://raw.githubusercontent.com/henryju/image-hosting/master/info.png)Do not forget to remove this deprecated code someday.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)